pkg/reconciler/v1alpha1/revision/resources/deploy_test.go

```diff
@@ -619,40 +545,32 @@ func TestMakePodSpec(t *testing.T) {
 		),
 	}, {
 		name: "complex pod spec",
-		rev: &v1alpha1.Revision{
-			ObjectMeta: metav1.ObjectMeta{
-				Namespace: "foo",
-				Name:      "bar",
-				UID:       "1234",
-			},
-			Spec: v1alpha1.RevisionSpec{
-				ContainerConcurrency: 1,
-				Container: corev1.Container{
-					Image:   "busybox",
-					Command: []string{"/bin/bash"},
-					Args:    []string{"-c", "echo Hello world"},
-					Env: []corev1.EnvVar{{
-						Name:  "FOO",
-						Value: "bar",
-					}, {
-						Name:  "BAZ",
-						Value: "blah",
-					}},
-					Resources: corev1.ResourceRequirements{
-						Requests: corev1.ResourceList{
-							corev1.ResourceMemory: resource.MustParse("666Mi"),
-							corev1.ResourceCPU:    resource.MustParse("666m"),
-						},
-						Limits: corev1.ResourceList{
-							corev1.ResourceMemory: resource.MustParse("888Mi"),
-							corev1.ResourceCPU:    resource.MustParse("888m"),
-						},
+		rev: revision(
+			withContainerConcurrency(1),
+			func(revision *v1alpha1.Revision) {
+				revision.ObjectMeta.Labels = map[string]string{}
+				revision.Spec.Container.Command = []string{"/bin/bash"}
+				revision.Spec.Container.Args = []string{"-c", "echo Hello world"}
+				revision.Spec.Container.Env = []corev1.EnvVar{{
```

Comment:
Use the env var option for this?

Reply:
Done, although I think it could be better.

The existing withEnvVar option takes a container, but here we have a Revision. In C++ I would overload the withEnvVar function, is there a good idiom for this in go?

Reply:
Not that I'm aware of...